src/frontend/components/SubscribeScheduledVideo/index.tsx

```diff
@@ -0,0 +1,182 @@
+import {
+  Box,
+  Clock,
+  Heading,
+  Notification,
+  Paragraph,
+  StatusType,
+} from 'grommet';
+import { DateTime } from 'luxon';
+import React, { useState, useRef } from 'react';
+import { Nullable } from '../../utils/types';
+import { defineMessages, FormattedMessage, useIntl } from 'react-intl';
+import { useAsyncEffect } from '../../utils/useAsyncEffect';
+import { Video } from '../../types/tracks';
+import { WaitingLiveVideo } from '../WaitingLiveVideo';
+import { SubscribeScheduledVideoEmailForm } from '../SubscribeScheduledVideoEmailForm';
+import { getDecodedJwt } from '../../data/appData';
+import { fetchList } from '../../data/queries/fetchList';
+/** Props shape for the SubscribeScheduledVideo component. */
+interface SubscribeScheduledVideoProps {
+  video: Video;
+}
+const timeIsOver = 'P0H0M0S';
+const durationFullDay = 'PT23H59M59S';
+
+const messages = defineMessages({
+  daysLeft: {
+    defaultMessage:
+      'Days left {daysLeft,number} {daysLeft, plural, =0 {day} one {day} other {days}}',
+    description:
+      'Text message to inform on number of days left before the event starts',
+    id: 'components.SubscribScheduledVideo.daysLeft',
+  },
+  register: {
+    defaultMessage: 'Register to this event starting on {date}',
+    description: 'Text message to ask if user is already registered ?',
+    id: 'components.SubscribScheduledVideo.register',
+  },
+  waitingMessage: {
+    defaultMessage:
+      'You are registered for this event. Live will start at {date}.',
+    description:
+      'Message reminding the user, he is already registered and giving back the starting at date.',
+    id: 'components.SubscribScheduledVideoEmail.waitingMessage',
+  },
+});
+
+export const SubscribeScheduledVideo = ({
+  video,
+}: SubscribeScheduledVideoProps) => {
+  const [daysLeft, setDaysLeft] = useState<Nullable<number>>(null);
+  const [timeLeft, setTimeLeft] = useState('');
+  const [isTimeOver, setTimeOver] = useState(false);
+  const [isAlreadyRegistered, setAlreadyRegistered] = useState(false);
+  const startingAt = useRef(new Date(video.starting_at!));
+  const intl = useIntl();
+
+  const initDays = () => {
+    const startingAtMs = DateTime.fromMillis(startingAt.current.getTime());
+    const diff = startingAtMs
+      .diffNow(['days', 'hours', 'minutes', 'seconds'])
+      .toObject();
+    // Using diff.toISO() would be better but Clock grommet component is not able to manage
+    // iso8601 period without hours.
+    // We need to truncate values
+    const seconds = Math.trunc(Number(diff.seconds!));
```

Comment:
IMO this is useless, `Clock` deals with it.

Reply:
If i haded it, it's because I had no choice to do it

Reply:
Ah, I don't really understand why.

The duration REGEX in the `Clock` component accepts float https://github.com/grommet/grommet/blob/97e68fb9134c3f9befdd5268d6119149f7973f80/src/js/components/Clock/Clock.js#L8-L9

Having `34.5` seconds in a Period is valid https://en.wikipedia.org/wiki/ISO_8601#Durations

I see no error when trying here https://codesandbox.io/s/grommet-v2-template-forked-m6vfd?file=/index.js

Can you explain me why ?

Reply:
I suppose then I only did it to have tests ok

Reply:
true I don't need it for seconds